debug_reg_printer.c

```diff
@@ -62,22 +62,34 @@ static uint64_t riscv_debug_reg_field_value(riscv_debug_reg_field_info_t field,
 
 static unsigned int riscv_debug_reg_fields_to_s(char *buf, unsigned int offset,
 	struct riscv_debug_reg_field_list_t (*get_next)(riscv_debug_reg_ctx_t contex),
-	riscv_debug_reg_ctx_t context, uint64_t value)
+	riscv_debug_reg_ctx_t context, uint64_t value,
+	enum riscv_debug_reg_show show)
 {
 	unsigned int curr = offset;
-	curr += get_len_or_sprintf(buf, curr, " { ");
+	curr += get_len_or_sprintf(buf, curr, " {");
+	char *separator = "";
 	for (struct riscv_debug_reg_field_list_t list; get_next; get_next = list.get_next) {
 		list = get_next(context);
-		curr += riscv_debug_reg_field_to_s(buf, curr, list.field, context,
-				riscv_debug_reg_field_value(list.field, value));
-		curr += get_len_or_sprintf(buf, curr, ", ");
+
+		uint64_t field_value = riscv_debug_reg_field_value(list.field, value);
+
+		if ((show == RISCV_DEBUG_REG_SHOW_ALL) ||
+				(show == RISCV_DEBUG_REG_HIDE_UNNAMED_0 &&
+					(field_value != 0 || list.field.values)) ||
```

Comment:
This will not hide an unnamed zero if there is a name for any other value (`list.field.values == {[0] = NULL, [1] = "some name"}`)

Reply:
Fixed, I think. I don't see any examples of this.

Reply:
> This will not hide an unnamed zero if there is a name for any other value (`list.field.values == {[0] = NULL, [1] = "some name"}`)

For an example, please, take a look at `csr_dcsr_cause_values` (`[0]` is zero-initialized by default).